### DISCUSSION TITLE
Use ImageMagick Libraries in C project

### DISCUSSION THREAD

**Discussion by lorenzopantano** [ImageMagick/ImageMagick#2604] [Help]
Hi, 
I'm building a C project and I need to use the ImageMagick libraries to compress images. I downloaded the GitHub repository, run configure, make, and make install. I'm new to this kind of thing so I don't know exactly which files I need to import in my project.
(Is there a way to use static libraries?).
Thanks for any help.

**Comment by snibgo** [comment]
You don't say what build platform you use, so advice is difficult.

[Compiling the MagickWand demonstration programs](http://im.snibgo.com/compdemo.htm) explains how I build programs that use IM, with the Gnu C toolset. The process is very simple.

If you use something else, such as MSVC, I suggest you search the IM forums.

**Comment by lorenzopantano** [comment]
I'm using gcc on a Linux subsystem in Windows, with a makefile. Thanks

**Comment by snibgo** [comment]
My pages show how to use gcc on Windows, with Cygwin bash. I expect the Linux subsystem is very similar. When I wrote those pages, the usual version of IM was v6. Now I use v7, but the process is the same.

**Comment by lorenzopantano** [comment]
Thanks for your reply, but I need more help. 
Consider I'm in a Linux environment, I need to use IM in my C project with a makefile. How can I achieve that? What I need to download? Where do I start? (I'm very confused about using the IM APIs)

**Comment by snibgo** [comment]
I suggest you start with the IM demonstration programs, such as core.c and wand.c. Work out the commands you need to build those programs. (I expect it will will need two commands: pkg-config and gcc, but you could combine them.) You may regard installation as a separate operation.

Then use that knowledge to build your own programs.

How you incorporate commands these in your makefile is up to you. But I suggest not to worry about that yet.
